cockroachdb/goose/migration.go

```diff
@@ -0,0 +1,43 @@
+package utils
+
+import (
+	"context"
+	"database/sql"
+	"runtime"
+
+	"github.com/jackc/pgx/v5"
+	"github.com/jackc/pgx/v5/stdlib"
+	"github.com/pressly/goose/v3"
+)
+
+type PgxContextMigration func(context.Context, *pgx.Conn) error
+
+func AddPgxContextMigration(upMigration PgxContextMigration, downMigration PgxContextMigration) {
+	// runtime.Caller returns the filename of the migration file that calls this function
+	_, filename, _, ok := runtime.Caller(1)
+	if !ok {
+		panic("Could not get caller filename")
```

Comment:
I'm not comfortable with a panic here. Can't we return an error?

Reply:
We could. 
However we are using this method from an init function  and we are not able to return an error.

Reply:
I will return an error for the moment

Reply:
IMO a library should _never_ panic. The calling method will deal the error the way it wants to deal with it. But the library shouldn't panic.

Reply:
Thanks !